src/general/IWorldEffect.cs

```diff
@@ -37,13 +38,24 @@ public void OnTimePassed(double elapsed, double totalTimePassed)
         {
             var patch = targetWorld.Map.Patches[key];
 
-            foreach (var compound in patch.Biome.Compounds.Keys)
+            // If there are microbes to be eating up the primordial soup, reduce the milk
+            if (patch.SpeciesInPatch.Keys.Count > 0)
```

Comment:
According to some documentation dictionary.Count should already return the number of keys in it.